csharp/tutorials/nullable-reference-migration/finished/SimpleFeedReader/Pages/Error.cshtml.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace SimpleFeedReader.Pages
+{
+    // <SnippetErrorModel>
+#nullable enable
+    public class ErrorModel : PageModel
+    {
+        public string? RequestId { get; set; }
+
+        public bool ShowRequestId => !string.IsNullOrEmpty(RequestId);
+
+        public void OnGet()
+        {
+            RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier;
+        }
+    }
+#nullable restore
```

Comment:
Do we genuinely expect people to use `#nullable restore` much? From a quick scan of the code, it looks like you're only doing it in places where it wouldn't make any difference anyway.
I suspect someone migrating file-by-file will just use `#nullable enable` at the top of the file and leave it at that. If there's no reason to discourage that (and there may be something I haven't thought of!) it's probably worth explicitly mentioning that - and that `#nullable restore` is only really needed when you want to migrate *part* of a file.

Reply:
Right. I'm leaning toward pairing enable / restore because it does the right thing even when a file has multiple classes. I think it might also help if a class contains nested types. 

I agree that the most common use case is turning it on at the beginning of a file and leaving it on.  I'll likely revisit this first tutorial once more have been written.